### ISSUE TITLE
model.learn() is running indefinitely irrespective of total_timesteps

### ISSUE THREAD

**Issue by FieteO** [DLR-RM/stable-baselines3#1404]
### 🐛 Bug

I have a custom gym environment that is passing the `check_env` method. However, when training it using `model.learn`, it is never finishing, going over the number of `total_timesteps` (the tqdm progress bar is showing 100/1 for instance)

```python
from stable_baselines3.common.evaluation import evaluate_policy
from stable_baselines3 import PPO

from custom_gyms.my_env.my_env import MyEnv

env = MyEnv()

model = PPO("MlpPolicy", env, verbose=1)
model.learn(total_timesteps=1, progress_bar=True)

mean_reward, std_reward = evaluate_policy(model, model.get_env(), n_eval_episodes=1)
print(f'mean_reward: {mean_reward}, std_reward: {std_reward}')
```

This is happening with and without returning `done = True` after each step

![output](https://user-images.githubusercontent.com/41323592/227212442-8d875af6-d98d-4be2-9d80-3810401e2245.png)

### Code example

```python
from typing import Any, List, Tuple, Type
import gym

import numpy as np
from gym import spaces
from gym.envs.registration import EnvSpec

class MyEnv(gym.Env):
    metadata = {'render.modes': ['console']}
    spec = EnvSpec("MyEnv-v0", entry_point='')

    def __init__(self):
        super(MyEnv, self).__init__()

        self.observation_space = spaces.Box(low=0.0, high=np.inf, shape=(60,15), dtype=np.float32)
        self.action_space = spaces.MultiBinary(60)
        self.reward = 0.0

    def get_obs(self):
        '''get the last n observations per cell'''
        print('get_obs()')
        return np.ones(60 * 15, dtype=np.float32).reshape(60,15)

    def reset(self):
        # Reset the reward
        print('reset()')
        return self.get_obs()

    def get_reward(self):
        print('get_reward')
        return 1.0

    def act(self) -> None:
        print('act()')
        pass
        
    def step(self, actions: np.ndarray) -> Tuple[List[Type[str]], int, bool, List[Any]]:
        self.act(actions)
        self.obs = self.get_obs()
        episodic_reward = self.get_reward()
        self.reward += episodic_reward
        return self.obs, self.reward, True, {}

    def render(self):
        pass

    def close(self):
        print('close')
        pass
```

### System Info

```sh
$ python -c 'import stable_baselines3 as sb3; sb3.get_system_info()'
- OS: Linux-5.4.0-144-generic-x86_64-with-glibc2.29 # 161-Ubuntu SMP Fri Feb 3 14:49:04 UTC 2023
- Python: 3.8.10
- Stable-Baselines3: 1.8.0a10
- PyTorch: 2.0.0+cu117
- GPU Enabled: False
- Numpy: 1.24.2
- Gym: 0.21.0
```

### Checklist

- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a minimal working example to reproduce the bug
- [X] I have checked my env using the env checker
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
> I have a custom gym environment that is passing the check_env method. However, when training it using model.learn, it is never finishing, going over the number of total_timesteps (the tqdm progress bar is showing 100/1 for instance)

Duplicate of https://github.com/DLR-RM/stable-baselines3/issues/1150
The training will finish here but only after `n_steps=2048` (default value of PPO, you can change `n_steps` hyperparameter if you want to change that behavior)

**Comment by araffin** [comment]
> Thanks! Wouldn't it make sense to print a warning then, that the total_timesteps parameter will be ignored? I find that rather confusing to let the users run into that unexpected behaviour...

The total timesteps is not ignored but adapted to fit how the algorithm works (see explanation) and most of the time the actual total timestep will be quite close to the one requested. As written in the README, despite being simple to use, SB3 assumes that the user has some knowledge about RL and the RL algorithms used (we have links for good resources in the doc).

However, I agree we should add a note in the documentation and for that we always welcome contributions ;)

**Comment by FieteO** [comment]
Okay, I will take a look at the docs next time before posting

**Comment by FieteO** [comment]
I see the same behaviour of running longer with `evaluate_policy(n_eval_episodes=1)`. However there, I cannot spot another parameter [in the docs](https://stable-baselines3.readthedocs.io/en/master/common/evaluation.html?highlight=evaluate_policy#module-stable_baselines3.common.evaluation) that could influence this. Is there also another default parameter in the background that is silently used instead @araffin ?

**Comment by FieteO** [comment]
Strange, for me it's running much longer in terms of steps.

Maybe the question is then more what constitutes an episode.
I am never returning `done=True` in my step function, since I have a continuous task that is returning a reward after each step and should not be reset as such. I would think it's non-episodic then.

**Comment by araffin** [comment]
>  I am never returning done=True in my step function, since I have a continuous task that is returning a reward after each step and should not be reset as such. I would think it's non-episodic then.

if you are never returning `done=True`, then it's never gonna stop (as it is running for what you ask, one episode, which is infinite in that case).
That's usually why we add a `TimeLimit` wrapper to infinite horizon tasks (we treat the timeout differently in SB3 as it's a special termination).

> observe the evaluation monitor returns one evaluation line more than usual ? Same for tqdm.

Please read the linked issues, yes same origin.
